_posts/2023-02-03_medical-imaging-with-nix/medical-imaging-with-nix.Rmd

```diff
@@ -0,0 +1,211 @@
+---
+title: Reproducible medical imaging software environments in Nix,</br> or,</br> Living in the future is hard.
+description: |
+  My journey creating a reproducible environment for building medical imaging machine learning models with Nix
+author:
+  - name: Chris Hammill
+date: 2023-01-27
+output:
+  distill::distill_article:
+    self_contained: false
+categories:
+  - post-miscellaneous
+draft: true
+#preview: this_is_fine.png
+---
+
+
+I want to tell a story, advertise a product, and gripe about technology all at the same time. For readers
+who don't know me, I'm Chris Hammill, a senior data scientist on Unity Health Toronto's Data Science and Advanced
+Analytics team. I've been thinking for a long time about how to build software environments that are resilient to the
+ravages of time and that others can pick up and use effectively at their leisure. So when in the course of my work
+at DSAA I needed to develop an effective environment for doing machine learning on medical imaging data, I jumped at
+the chance. The fruit of that labour is:
+
+https://github.com/LKS-CHART/medical-imaging-nix
+
+a working environment of software, defined in software. It allows me or anyone else to reproduce my work or use the
+exact software computing environment I use. What do I mean by environment? In this case I mean isolated collections of 
+programs, packages, and libraries that can be used together, but don't interact with the rest of your software. 
+The concept might be familiar if you've worked with [renv](https://posit.co/blog/renv-project-environments-for-r/), [conda](https://docs.conda.io/en/latest/), or [lmod]()https://lmod.readthedocs.io/en/latest/).
+
+The project is powered by [nix](https://nixos.org/), a futuristic technology gaining a foothold among certain groups of 
+software developers and computing enthusiasts. Renowned for its ability to almost fully deterministically and reproducibly 
+build software, enabling strongly isolated environments.
+
+## What is medical-imaging-nix?
+
+The repo above is a [nix](https://nixos.org/) flake that generates a software environment that can be used as a 
```

Comment:
Is a "flake" kind of like a container?

Reply:
I'm kind of blanking on how to describe it, it's a tool for composing nix builds. Say for example you have a build that depends on a specific version of nixpkgs and you want to use it with software that requires another specific version of nixpkgs. Flakes are a relatively new tool for encapsulating/importing entire projects that use nix. Maybe they're closest to nix libraries. I've avoided complicating the narrative by describing flakes, do you think I need to/should?

Reply:
Actually, the more I read the blog, the more it made sense. I don't think you need to describe it!

Reply:
ok sounds good, thanks Chloe!